### PR TITLE
11551 subscribe/unsubscribe events

### PR DESCRIPTION
**Pull request by mahsa-bytt**
related PR:
https://github.com/quantnetwork/overledger-bpi-gateway-service/pull/28

### CODE REVIEW

**Review by theoQuant — APPROVED** [review]

File: overledger-sdk-essential/src/main/java/network/quant/essential/OverledgerClient.java [hunk 1]
```diff
@@ -140,6 +141,69 @@ public ContractQueryResponseDto smartContractQuery(DltTransaction dltTransaction
...
+                    .onStatus(HttpStatus::is4xxClientError, this::getClientResponse)
+                    .onStatus(HttpStatus::is5xxServerError, this::getClientResponse)
+                    //.onStatus(HttpStatus::is2xxSuccessful, this::successMSG)
+                    .onStatus(HttpStatus::is3xxRedirection, clientResponse -> Mono.error(new RedirectException(clientResponse.headers().header(HEADER_LOCATION).get(0))))
+                    .bodyToFlux(EventSubscribeResponse.class)
+                    .timeout(Duration.ofMillis(120000))
+                    .collectList()
+                    .block();
+        } catch (RedirectException e) {
+            return this.webClient
```
Comment (richardbahari) [inline overledger-sdk-essential/src/main/java/network/quant/essential/OverledgerClient.java:162]: just a minor comment, with the redirect, we should probably do the same timeout condition

Comment (mahsa-bytt) [reply overledger-sdk-essential/src/main/java/network/quant/essential/OverledgerClient.java:162]: resolved & pushed.

File: overledger-sdk-ethereum/src/main/java/network/quant/ethereum/experimental/ContractArgumentToAbiMainFactory.java [hunk 2]
```diff
@@ -88,6 +89,32 @@ public static Type convertContractArgument(ContractArgument contractArgument) {
         return null;
     }
 
+    public static TypeReference generateTypeReference(ContractArgument contractArgument){
```
Comment (richardbahari) [inline overledger-sdk-ethereum/src/main/java/network/quant/ethereum/experimental/ContractArgumentToAbiMainFactory.java:92]: when do we use this method by the way? i couldn't find the usage on intellij

Comment (mahsa-bytt) [reply overledger-sdk-ethereum/src/main/java/network/quant/ethereum/experimental/ContractArgumentToAbiMainFactory.java]: This method is now in common-message project and we don't use them here, I removed them.

**Review by richardbahari — APPROVED** [review]
looks ok, perhaps have Najla and/or Luke also to take a look

**Comment by najla-quant** [comment]
@mahsa-bytt @richardbahari  Yes, I would take a look but tomorrow :)

**Comment by najla-quant** [comment]
@mahsa-bytt The code looks good. I still have to run the corresponding example on testnet. I am getting Client Response error at the moment. Could you please run it locally on your side on testnet with:

overledger.bpikey=joNp29bJkQHwEwP3FmNZFgHTqCmciVu5NYD3LkEtk1I
overledger.mappid=network.quant.testnet

To see if the source problem is my config or the service.

**Review by najla-quant — APPROVED** [review]

**Pull request merged**